transformer_engine/pytorch/tensor/float8_tensor.py

```diff
@@ -208,20 +207,25 @@ def __init__(
         columnwise: bool = True,
         with_amax_reduction: bool = False,
         amax_reduction_group: Optional[dist_group_type] = None,
-        amax_reduction_size: Optional[int] = 1,
         force_pow_2_scales: bool = False,
         amax_epsilon: float = 0.0,
+        amax_reduction_size: Any = None,  # deprecated
     ) -> None:
         super().__init__(rowwise=rowwise, columnwise=columnwise)
         self.scale = torch.ones(1, dtype=torch.float32, device=device)
         self.amax = torch.empty(1, dtype=torch.float32, device=device)
         self.dtype = fp8_dtype
         self.with_amax_reduction = with_amax_reduction
         self.amax_reduction_group = amax_reduction_group
-        self.amax_reduction_size = amax_reduction_size
         self.force_pow_2_scales = force_pow_2_scales
         self.amax_epsilon = amax_epsilon
 
+        # Deprecated option
+        if amax_reduction_size is not None:
+            warnings.warn(
+                "amax_reduction_size kwarg is deprecated", DeprecationWarning, stacklevel=2
```

Comment:
If we are going to merge this in 2.2 then we don't need to deprecate it since it never existed in a stable release - we can just remove it.